llarp/iwp/session.cpp

```diff
@@ -254,20 +254,17 @@ namespace llarp
           }
         }
       }
-      auto self = shared_from_this();
       if(m_EncryptNext && !m_EncryptNext->empty())
       {
-        m_Parent->QueueWork([self, data = std::move(m_EncryptNext)] {
-          self->EncryptWorker(data);
-        });
+        m_Parent->QueueWork(std::bind(&Session::EncryptWorker,
+                                      shared_from_this(), m_EncryptNext));
         m_EncryptNext = nullptr;
```

Comment:
This assignment seems suspect to me, but also unnecessary with the previous code: the `data = std::move(m_EncryptNext)` is required to leave `m_EncryptNext` empty.

Reply:
Do any other threads ever touch `m_EncryptNext`?

Reply:
no other threads *should be* touching `m_EncryptNext`